server_test.go

```diff
@@ -763,6 +763,112 @@ func TestServerResponseBodyStream(t *testing.T) {
 	}
 }
 
+func TestServerResponseBodyStreamProxy(t *testing.T) {
+	t.Parallel()
+
+	ln := fasthttputil.NewInmemoryListener()
+	lnProxy := fasthttputil.NewInmemoryListener()
+
+	h := func(ctx *RequestCtx) {
+		ctx.SetConnectionClose()
+		body := []byte("body")
+		ctx.SetBodyStream(bytes.NewBuffer(body), len(body))
+	}
+
+	hProxy := func(ctx *RequestCtx) {
+		ctx.SetConnectionClose()
+
+		cli := Client{
+			Dial: func(addr string) (net.Conn, error) {
+				return lnProxy.Dial()
+			},
+		}
+		err := cli.Do(&ctx.Request, &ctx.Response)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		responseBodyStream := ctx.ResponseBodyStream()
+		ctx.Response.Body()
+		body, _ := io.ReadAll(responseBodyStream)
+		body = append(body, []byte(" append")...)
+		ctx.SetBodyStream(bytes.NewBuffer(body), len(body))
+	}
+
+	serverCh := make(chan struct{})
+	go func() {
+		if err := Serve(lnProxy, h); err != nil {
+			t.Errorf("unexpected error: %v", err)
+		}
+		close(serverCh)
+	}()
+
+	serverProxyCh := make(chan struct{})
+	go func() {
+		if err := Serve(ln, hProxy); err != nil {
```

Comment:
Shouldn't your test set [`Server.StreamRequestBody = true`](https://pkg.go.dev/github.com/valyala/fasthttp#Server.StreamRequestBody)? That's what the test should be testing, if that works correctly. Otherwise it's isn't testing something new as it just tests the `bytes.NewReader(ctx.Response.bodyBytes())` line of your function which is trivial.